src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs

```diff
@@ -104,29 +104,26 @@ public JsonResult Create(TeamViewModel teamViewModel)
                     {
                         var newAddedCaptain = _playerService.Create(teamViewModel.Captain.ToCreatePlayerDto());
                         teamViewModel.Captain.Id = newAddedCaptain.Id;
+
                     }
 
+                    teamViewModel.Roster = new List<PlayerNameViewModel> {
+                        new PlayerNameViewModel {
+                            Id = teamViewModel.Captain.Id,
+                            FirstName = teamViewModel.Captain.FirstName,
+                            LastName = teamViewModel.Captain.LastName
+                        }
+                    };
+
+                    var playersIdToAddToTeam = SeparatePlayers(teamViewModel);
+
                     var team = teamViewModel.ToCreateTeamDto();
                     var createdTeam = _teamService.Create(team);
                     teamViewModel.Id = createdTeam.Id;
 
-                    createdTeam.SetCaptain(new PlayerId(createdTeam.Captain.Id));
                     _playerService.AssingPlayerToTeam(teamViewModel.Captain.ToDomain(), createdTeam.Id);
 
-                    if (teamViewModel.AddedPlayers.Count > 0)
-                    {
-                        var playersToAddToTeam = _playerService.CreateBulk(teamViewModel.AddedPlayers
-                                .Where(x => x.Id == 0)
-                                .Select(x => x.ToCreatePlayerDto())
-                                .ToList());
-
-                        var playersIdToAddToTeam = playersToAddToTeam
-                                .Select(x => new PlayerId(x.Id))
-                                .ToList();
-
-                        playersIdToAddToTeam.AddRange(teamViewModel.AddedPlayers.Where(x => x.Id > 0).Select(x => new PlayerId(x.Id)));
-                        _teamService.AddPlayers(new TeamId(teamViewModel.Id), playersIdToAddToTeam);
-                    }
+                    _teamService.AddPlayers(new TeamId(teamViewModel.Id), playersIdToAddToTeam);
```

Comment:
Why do we need this one?

Reply:
Do you meen  _teamService.AddPlayers?  Add new or existing player to roster, as we add captain?